### PR TITLE
Fixes a few things potentially preventing hijack objectives

### LINKED ISSUE #13705: revenant's count a loyal crewmember for hijack
Yeah, you fail hijack when a revenant is on board. Doesn't really make sense that they would count as a loyal crewmember for the objective. Kinda sucks because they can just go invisible.

### PR DESCRIPTION
**Pull request by ChangelingRain**
Including; a ranged holoparasite scouting from an escape pod onto the shuttle, revenants spawned improperly, brainmobs like MMIs and such(which possibly count even if from gibbing? I'm not sure, safer to just ignore them anyway)

Fixes #13705

If #17745 was a bug and not a stowaway, probably fixes it.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Question; should simple animals even really count?

**Comment by TheRealGLH** [comment]
I can't even think of a single simple animal that you could consider part of the crew, except MAYBE a headslugged ling and even that's a stretch.

**Comment by Fox-McCloud** [comment]
Exactly, it also means you can't do wild sentience potion hijacks either.

**Comment by Dorsisdwarf** [comment]
Beepsky's a simple mob and he is the LAW

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -224,8 +224,15 @@
 	for(var/mob/living/player in player_list)
 		if(player.mind && player.mind != owner)
 			if(player.stat != DEAD)
-				if(istype(player, /mob/living/silicon)) //Borgs are technically dead anyways
-					continue
+				switch(player.type)
```
Comment (RemieRichards) [inline code/game/gamemodes/objective.dm]: No, don't ```switch()``` types

**Comment by Fox-McCloud** [comment]
>Beepsky's a simple mob and he is the LAW

He's a simple mob only in terms of code--in practicality, he's a bot and absolutely should not count (nor did he ever until recent times).

**Pull request merged**